content/python/concepts/modules/terms/pytorch/concepts/autograd/autograd.md

```diff
@@ -0,0 +1,203 @@
+
+To compute gradients, autograd forward passes all input values in the `Tensor` to calculate the loss. Then in the backward pass (also known as backpropagation), the gradients are calculated to generate graphs. In backpropagation, gradients are computed to discover if adjustments need to be made to minimize loss in the learning model. This allows autograd to update learning models' weight values during training and provides a method for computing gradients by using scalar values (also known as floating point).
+
+>[!Note]
+> Only tensors of floating point and complex dtype can have gradients.
+
+## Syntax
+To enable autograd in a `Tensor`, the gradient parameter needs to be set to `true`.
+
+`requires_grad=true`
+
+>[!Note] 
+> When the autograd parameter is set to `requires_grad=false`, gradients cannot be computed and the graphs will return empty. This is because the backpropagation function required for calculating gradients is disabled. 
+
+
+## Using autograd to calculate gradients
+
+The following is a simple example of how to use autograd to calculate gradients.
+
+```python
+import torch 
+
+#Create a tensor and set the autograd parameter to true.
+x = torch.ones(3,3,3,requires_grad=True)
+print (x)
+
+""" 
+Output:
+tensor([[[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]],
+
+        [[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]],
+
+        [[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]]], requires_grad=True)
+"""
+
+#Print out the data for x.
+print (x.data)
+
+""" 
+Output:
+tensor([[[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]],
+
+        [[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]],
+
+        [[1., 1., 1.],
+         [1., 1., 1.],
+         [1., 1., 1.]]])
+"""
+```
+Note that at this point no gradient functions have been created yet, so `print (x.grad)` will not work. To solve this issue, create a gradient function.
+
+```python
+
+#Create a gradient function by specifying the operation for x.
+y = x * 3
+print(y)
+
+"""
+Output:
+tensor([[[3., 3., 3.],
+         [3., 3., 3.],
+         [3., 3., 3.]],
+
+        [[3., 3., 3.],
+         [3., 3., 3.],
+         [3., 3., 3.]],
+
+        [[3., 3., 3.],
+         [3., 3., 3.],
+         [3., 3., 3.]]], grad_fn=<MulBackward0>)
+
+"""
+```
+Because *y* was created from the operation, it now has a gradient function.
+
+```python
+#Print out the created gradient function.
+print (y.grad_fn)
+
+""" 
+Output:
+MulBackward0 object at 0x11f5b2950
+"""
+```
+
+By default, the Pytorch gradient computations clear all internal buffers in the graph. To perform backpropagation twice on parts of the graph, parameters must be passed in `retain_variables = True` during the forward pass. If the retained variables were not specified at this step, certain operations will fail later on.
+
+```python
+x = torch.ones(3,3,3, requires_grad=True)
+y = x * 3
+y.backward(torch.ones(3,3,3), retain_graph=True)
+print(x.grad)
```

Comment:
Hi @SaviDahegaonkar,
I removed "retain_graph=True" because as you said it is only applied when the backward method needs to be passed backward multiple times on the same graph.